WooCommerce/Classes/Model/MarkOrderAsReadUseCase.swift

```diff
@@ -1,30 +1,79 @@
 import Foundation
+
+#if canImport(Networking)
+import Networking
+#elseif canImport(NetworkingWatchOS)
+import NetworkingWatchOS
+#endif
+
+#if canImport(Yosemite)
 import Yosemite
+#endif
 
 struct MarkOrderAsReadUseCase {
+    /// Possible error states.
+    ///
+    enum Error: Swift.Error {
+        case failure(Swift.Error)
+        case updateReadStatus(Swift.Error)
+        case unavailableNote
+    }
     /// Method that marks the order note as read if it is the notification for the last order.
     /// We do it in a way that first we syncronize notification to get the remote `Note`
     /// and then we compare local `orderID` with the one from remote `Note`.
     /// If they match we mark it as read.
     /// We pass syncronized note and error in the `onCompletion` completion block
+#if canImport(Yosemite)
     static func markOrderNoteAsReadIfNeeded(stores: StoresManager, noteID: Int64, orderID: Int, onCompletion: ((Note?, Error?) -> Void)? = nil) {
```

Comment:
Updates for **MarkOrderAsReadUseCase** to have two versions of **markOrderNoteAsReadIfNeeded**, one usses **NotificationsRemote** and the second one uses **NotificationAction**. That way we can have same functionality in the iOS app, notification extension and watchOS app.

Good question would be why not just use **NotificationsRemote** in all places. When we use **NotificationAction** we can see that **NotificationStore** in **synchronizeNotification**() function also calls **updateLocalNotes**(with: notes) and that is a useful addition worth having 2 implementations.